### PR TITLE
refactor(Request): Implement `getServerName()` method with PSR-7 support and fallback to Yii2, including comprehensive tests.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

## Summary by CodeRabbit

* **New Features**
  * Added support for retrieving the server name from requests, with improved handling of edge cases.
* **Bug Fixes**
  * Enhanced error handling for missing server variables, ensuring clear exceptions are raised when necessary.
* **Tests**
  * Expanded test coverage for server name retrieval, cookie validation, and error scenarios.
  * Improved assertion accuracy and consistency in test messages.
  * Simplified test cleanup and global state management for better reliability.
* **Chores**
  * Introduced a constant for cookie validation keys to standardize test configurations.

### CODE REVIEW

File: tests/http/RequestTest.php [hunk 1]
```diff
@@ -1698,6 +1719,30 @@ public function testThrowExceptionWhenRequestUriIsMissing(): void
...
+    {
+        $request = new Request();
+
+        $_SERVER = [];
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('Unable to determine the entry script file path.');
+
+        $request->getScriptFile();
+    }
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply tests/http/RequestTest.php]: fixed.

File: tests/adapter/ServerRequestAdapterTest.php [hunk 2]
```diff
@@ -2111,50 +2078,174 @@ public function testSecureHeadersAreFilteredWhenNotFromTrustedHost(): void
...
+            [
+                'another_cookie' => 'another_value',
+                'test_cookie' => 'test_value',
+            ],
+        );
+
+        $request = new Request();
+
+        $request->enableCookieValidation = false;
+        $request->cookieValidationKey = 'test-validation-key-32-characters';
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply tests/adapter/ServerRequestAdapterTest.php:2215]: fixed.

File: tests/TestCase.php [hunk 3]
```diff
@@ -20,6 +20,11 @@
 
 abstract class TestCase extends \PHPUnit\Framework\TestCase
 {
+    /**
+     * A secret key used for cookie validation in tests.
+     */
+    protected const COOKIE_VALIDATION_KEY = 'wefJDF8sfdsfSDefwqdxj9oq';
+
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply tests/TestCase.php:27]: They are tests so it's fine for me, it's a random string.
